File: backend/src/catalog/core/ProductImage.ts
/**
 * Represents a ProductImage domain entity.
 */
export class ProductImage {
  constructor(
    public readonly id: string,
    public readonly productId: string,
    public readonly url: string,
    public readonly position: number,
    public readonly createdAt: string,
  ) {}

  public static create(params: {
    id: string;
    productId: string;
    url: string;
    position?: number;
    createdAt?: string;
  }): ProductImage {
    return new ProductImage(
      params.id,
      params.productId,
      params.url,
      params.position ?? 0,
      params.createdAt ?? new Date().toISOString(),
    );
  }
}

File: mobile/lib/features/catalog/domain/entities/product_image.dart
/// ProductImage domain entity on mobile.
class ProductImage {
  final String id;
  final String productId;
  final String url;
  final int position;

  const ProductImage({
    required this.id,
    required this.productId,
    required this.url,
    required this.position,
  });
}
